plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,260 @@
+package io.jenkins.plugins.casc.misc;
+
+import com.gargoylesoftware.htmlunit.WebRequest;
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.util.NameValuePair;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.FileSystemException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.logging.Level;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import static com.gargoylesoftware.htmlunit.HttpMethod.POST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ *
+ */
+public abstract class ExportImportRoundTripAbstractTest {
+    private String jenkinsConfigContentOld;
+    private String cascJenkinsConfigPropOld;
+
+    @Rule
+    public RestartableJenkinsRule r = new RestartableJenkinsRule();
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Rule
+    public LoggerRule logging = new LoggerRule();
+
+    /**
+     * A method to assert if the configuration was correctly loaded. The Jenkins rule and the content of the config
+     * supposedly loaded are passed.
+     */
+    public abstract void assertConfiguredAsExpected(RestartableJenkinsRule j, String configContent);
+
+    /**
+     * Return the resource path (yaml file) to be loaded. i.e: If the resource is in the same package of the implementor
+     * class, then: my-config.yaml
+     * @return the resource name and path.
+     */
+    public abstract String configResource();
+
+    /**
+     * Return the string that should be in the logs of the JCasC logger to verify it's configured after a restart. This
+     * string should be unique to avoid interpreting that it was configured successfully, but it wasn't.
+     * @return the unique string to be in the logs to certify the configuration was done successfully.
+     */
+    public abstract String stringInLogExpected();
+
+    @Before
```

Comment:
I think it can be dangerous. If someone extends this class and adds other tests that are not related with the subject of this abstract class, this method would be executed always. Maybe it's worth to call it from the test method